primitives/direct_rpc_server/src/rpc_watch_extractor.rs

```diff
@@ -0,0 +1,123 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{DetermineWatch, DirectRpcError, DirectRpcResult};
+use codec::Decode;
+use std::marker::PhantomData;
+use substratee_worker_primitives::{DirectRequestStatus, RpcResponse, RpcReturnValue};
+
+pub struct RpcWatchExtractor<Hash>
+where
+	Hash: std::hash::Hash + Eq + Decode,
+{
+	phantom_data: PhantomData<Hash>,
+}
+
+impl<Hash> RpcWatchExtractor<Hash>
+where
+	Hash: std::hash::Hash + Eq + Decode,
+{
+	pub fn new() -> Self {
+		Self::default()
+	}
+}
+
+impl<Hash> Default for RpcWatchExtractor<Hash>
```

Comment:
It would require an additional trait bound on Hash: `Hash: Default` - which I'm not sure I want to impose?
But regardless, I don't find it ideal here, I couldn't find a way around the `PhantomData`, because the type parameter `Hash` is only used in the implementation of the `DetermineWatch` trait, but isn't relevant in the rest of the struct.. I don't know what the nice solution would be here..